lib/datadog/ci/contrib/rspec/example.rb

```diff
@@ -49,13 +54,25 @@ def run(*)
                   case execution_result.status
                   when :passed
                     test_span.passed!
+                    test_suite_span.passed! if test_suite_span
```

Comment:
I think you should just force `test_suite_span` to exist 😄  (and make it some sort of error condition if it doesn't).

Reply:
I tried to do that and it made the inheritance hierarchy unsound (so I needed to provided Null* subclass for every class that was cumbersome or always assume that superclass is returned which breaks Liskov substitution principle, see some more context in https://github.com/DataDog/datadog-ci-rb/pull/108 - look for NullSpan explanation).

Thankfully, I will be saved from this unruly ifs when we decouple ourselves from dd-trace-rb! I will immediately migrate this project to Ruby 2.7 syntax that will make these calls just:

```ruby
test_suite_span&.passed!
```

Neat, isn't it?